---
title: "Methods-Efficient Translation of Autism Genetics Research for Clinical Decision Support"
author: "Olivia J. Veatch"
date: "November 30, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
***
### Methods
1) To identify an initial list of candidate genes for ASD, data available in DisGeNET (http://www.disgenet.org) was queried. The benefits of this database include that the reported gene-disease associations' are identified by text-mining data from multiple sources and the data can be automatically pulled via the RCurl package in R. All of the Unified Medical Language System (UMLS) Concept Unique Identifiers (CUI) that relate to ASD (e.g., Autism Spectrum Disorder, Atypical autism, Autistic Behavior, and utistic Disorder) were identified from the disease mappings file provided by the database. An R function provided by DisGeNET was used to create a data frame of all genes with any level of evidence for a relationship with CUIs relating to ASD.
``` {r eval=TRUE}
library(RCurl)
library(tidyverse)
#download UMLS CUI to several disease categories file from http://www.disgenet.org/web/DisGeNET/menu/downloads#r and read into R environment

diseasecuis<-read.table(url("https://github.com/veatcho/EffGeneXlaCDS/blob/master/disease_mappings.tsv?raw=true"), header = TRUE, sep = "\t", fill = TRUE, stringsAsFactors = FALSE, quote="")

#Pull Autism-related CUIs
ASDcuis<-diseasecuis %>%
  select(diseaseId, name) %>%
  filter(grepl("^Autis|MACROCEPHALY/AUTIS|^AUTIS", name)) %>%
  distinct(diseaseId)

#Pulled original function from http://www.disgenet.org/ds/DisGeNET/scripts/disgenet.R and modified to suit my pipeline_OJV

# Copyright [2010-2017] Integrative Biomedical Informatics Group, Research Programme on Biomedical Informatics (GRIB) IMIM-UPF 
# http://ibi.imim.es/
# contact for technical questions support@disgenet.org
# creator: janet.pinero@upf.edu  
# Script to query disgenet using a list of genes or diseases
# requires as input the gene or disease list in a file 
# the output file name
# the type of entity (gene or disease)
# the type of identifier 
###############################################################################


# main
###############################################################################
# load packages  	
#require(RCurl)

###############################################################################
# subs
###############################################################################

doQuery_modifiedOJV = function(inputFile, outFile, entity, identifier){
  #print(inputFile)
  #print(outFile)
  #print(entity)
  #print(identifier)
  
  
  # read in all data
  #inFile = read.csv(file=paste(getwd(), inputFile, sep="/"), sep="\t", header=F)
  inFile = inputFile ###OJV altered to address the object from the previous command (ASDcuis), instead of loading a file that was saved on disk
  dataFin <- data.frame(matrix(nrow=0, ncol=14)) 
  
  STR = "";
  if (entity == "gene"){
    if (identifier == "entrez"){
      STR = "c2.geneId = '"
    }
    else  if (identifier == "hgnc"){
      STR = "c2.symbol = '"
    }
    else{
      stop ( "the type of identifier must be entrez gene identifiers or gene symbols \n")
    }
  }
  else if (entity == "disease"){
    if (identifier == "cui"){
      STR = "c1.diseaseId = '"
    }
    else  if (identifier == "mesh"){
      STR = "c1.MESH = '"
    }
    else  if (identifier == "omim"){
      STR = "c1.OMIM = '"
    }
    
    else{
      stop  ("the type of identifier must be cui or mesh or omim identifiers\n")
    }
  }
  else{
    stop ("the type of entity must be disease or gene \n");
  }
  #for (ent in inFile$V1 ){
  for (ent in inFile[,1] ){ ### OJV altered to be universal and pull 1st column by index as opposed to column name
    url <- "http://www.disgenet.org/oql"
    oql <- paste( "DEFINE
   	c0='/data/gene_disease_summary',
	c1='/data/diseases',
	c2='/data/genes',
	c4='/data/sources'
    ON
    'http://www.disgenet.org/web/DisGeNET'
    SELECT
	c1 (diseaseId, name, STY, MESH, OMIM, type),
	c2 (geneId, symbol,uniprotId, description, DPI, DSI, pantherName),
	c0 (score, EI, Npmids, Nsnps)

    FROM
    c0
    WHERE
    (
    c4 = 'ALL'
    AND ", STR, ent , "' )
    ORDER BY
    c0.score DESC" , sep = "")

    dataTsv <-  getURLContent(url, readfunction =charToRaw(oql), upload = TRUE, customrequest = "POST")
    #dataTsv <- rawToChar( getURLContent(url, readfunction =charToRaw(oql), upload = TRUE, customrequest = "POST"))
    myTextConnection <- textConnection( dataTsv ) 
    data <-  read.csv( myTextConnection, header = TRUE, sep = "\t" )
    close(myTextConnection)

    if (dim(data)[1] == 0 ){
      print ( paste (entity , ent, " is not in DisGeNET ", sep = " "))
    }
    else  {
      data$c0.EI <- ifelse(data$c0.EI == "null", NA, as.character(data$c0.EI) )
      dataFin <- rbind(dataFin, data)
    }
  
  }
  
  #Create outFile if it does not exist
  address <-  paste(getwd(), outFile, sep="/")

  print(address)
  
  write.table(dataFin,  address, sep="\t", row.names = F,dec = ".", quote = FALSE)
  return(dataFin) ###OJV altered to create dataframe in R environment
}

ASDcuigenes<-doQuery_modifiedOJV(ASDcuis, "ASDcuigenes.txt", "disease", "cui") ### NOTE: input file is object from previous command prior to doQuery function call

#Remove dataframes from environment that are no longer neccesary
rm(diseasecuis)
rm(ASDcuis)
```

2) Next, the list of ASD candidate genes was evaluated to identify genes that are specifically expressed in the tissue of most interest to ASD etiology (i.e., the human brain). Details for identifying genes with baseline expression in human tissue from GTEx are provided here: https://www.ebi.ac.uk/gxa/help/r-data-objects.html, here http://www.bioconductor.org/packages/release/bioc/html/ExpressionAtlas.html, and here https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-5214/. Optional code is provided below if the user prefers. As the direct pull for all GTEx data is large and we were only interested in brain regions, included in the Genotype Tissue Expression (GTEx) project were downloaded from the Expression Atlas (https://www.ebi.ac.uk/gxa/home) and the resulting GTEx data frame was filtered for ASD risk genes.
``` {r eval=TRUE}
# To download GTEx data in entirety
#if (!requireNamespace("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#BiocManager::install("ExpressionAtlas", version = "3.8")
#library('ExpressionAtlas')
#GTExdata<-getAtlasData("E-MTAB-5214")
GTExbraindata <- read.table(url("https://github.com/veatcho/EffGeneXlaCDS/blob/master/E-MTAB-5214-query-resultsbrain_TPM.tsv?raw=true"), header = TRUE, sep = "\t", fill = TRUE, stringsAsFactors = FALSE)

#Create dataframe with ASD candidate genes expressed in human brain
ASDcuigenes <- ASDcuigenes %>%
  select(c2.geneId, Gene.Name = c2.symbol, c2.uniprotId, c0.score) %>%
  mutate(Gene.Name = as.character(Gene.Name))
ASDcuigenesdistinct <- distinct(ASDcuigenes, c2.uniprotId, .keep_all = TRUE)
ASDbraingenes <- left_join(ASDcuigenesdistinct, GTExbraindata, by='Gene.Name')
ASDbraingenes <- select(ASDbraingenes, EntrezGeneID=c2.geneId, Gene.Name, UniProtID=c2.uniprotId, DisGeNETscore=c0.score, EnsID.GTEx=Gene.ID, Brodman1909.area24=Brodmann..1909..area.24, Brodmann1909.area9=Brodmann..1909..area.9, Amygdala=amygdala, Caudate.Nucleus=caudate.nucleus, Cerebellar.Hemisphere=cerebellar.hemisphere, Cerebellum=cerebellum, Cerebral.Cortex=cerebral.cortex, Hippocampus.Proper=hippocampus.proper, Hypothalamus=hypothalamus, Nucleus.Accumbens=nucleus.accumbens, Pituitary.Gland=pituitary.gland, Putamen=putamen, Substantia.Nigra=substantia.nigra)

#Remove dataframes from environment that are no longer neccesary
rm(GTExbraindata)
rm(ASDcuigenes)
rm(ASDcuigenesdistinct)
```

3) Run gene set analysis using TopGO to look for Gene Ontology-defined biological processes with overrepresentation of ASD risk genes
``` {r eval=TRUE}
#To install BiomaRt uncomment below
#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("biomaRt", version = "3.8")
#Following load of biomaRt package need to specify dplyr::select due to loading of AnnotatioDbi

#Pull names for all protein coding genes known in humans that are included in Ensembl
library(biomaRt)

humangenes <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
#listFilters(humangenes)
#listAttributes(humangenes)
ensIDswithproteins<-getBM(attributes=c("ensembl_gene_id", "entrezgene", "protein_id", "external_gene_name"), filters = list('with_protein_id'=TRUE), mart=humangenes)
ensIDswithproteins_distinctgenes <- distinct(ensIDswithproteins, entrezgene, .keep_all=TRUE)

#Remove data from environment that are no longer neccesary
rm(ensIDswithproteins)
rm(humangenes)

#Convert risk gene symbols to Ensembl IDs and create genes of interest list
ensIDswithproteins_distinctgenes <- ensIDswithproteins_distinctgenes %>%
  mutate(EntrezGeneID=entrezgene)

ASDensIDs_gene.names <- inner_join(ASDbraingenes[, c(1)], ensIDswithproteins_distinctgenes, by="EntrezGeneID")
ASDensIDs <- ASDensIDs_gene.names %>%
  distinct(EntrezGeneID, .keep_all=TRUE) %>%
  dplyr::select(ensembl_gene_id)

#Run GSA of disorder-associated genes to identify Gene Ontoloy-defined biological processes with overrepresentation compared to all human genes
#To Install TopGO uncomment:
#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("topGO", version = "3.8")
#To install annotations for human: uncomment
#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("org.Hs.eg.db", version = "3.8")

library(topGO)

all_genes <- ensIDswithproteins_distinctgenes[,1]
geneUniverse <- rep(0,length(all_genes))
names(geneUniverse) <- all_genes
genesOfInterest <- ASDensIDs[,1]
names(ASDensIDs) <- ASDensIDs
geneUniverse[grepl(paste(genesOfInterest, collapse = "|"), names(geneUniverse))] <- 1

ASD_GOBPdata<-new("topGOdata",
              description="testdata",
              ontology="BP",
              allGenes=geneUniverse,
              geneSel=function(p) p == 1, annot=annFUN.org, ID="ensembl", mapping="org.Hs.eg.db")

#Run Overrepresentation Test
ASDgene_BPGSA_classicfisher <- runTest(ASD_GOBPdata, algorithm = "classic", statistic = "fisher")

#Create data.frame with overrepresented GO-defined Biological processes for comparison with mouse KO phenotypes
ASDgene_BPGSA_classicfisher.results <- GenTable(ASD_GOBPdata, classicFisher = ASDgene_BPGSA_classicfisher, topNodes = 100)

ASDgeneBPs <- filter(ASDgene_BPGSA_classicfisher.results, grepl("< 1e-30",  classicFisher))

#To pull gene alias' that are assigned to significantly enriched GO terms
ASDGenes.GOIDs <- c(ASDgeneBPs$GO.ID)
Genes.in.SigGOIDs <- AnnotationDbi::select(org.Hs.eg.db, keys=ASDGenes.GOIDs, columns=c("ALIAS", "ENSEMBL", "ENTREZID"), keytype="GOALL")
Genes.in.SigGOIDs <- mutate(Genes.in.SigGOIDs, EntrezGeneID=as.integer(ENTREZID))
ASDGenes.in.SigGOIDs <- left_join(ASDbraingenes, Genes.in.SigGOIDs, by="EntrezGeneID")

#Determine to how many different overrepresented GOBPs each gene is assigned
ASDGenes.Brain.SigBPs <- ASDGenes.in.SigGOIDs %>%
  group_by(EntrezGeneID) %>%
  mutate(nGOBPsassigned = n_distinct(GOALL)) %>%
  distinct(EntrezGeneID, .keep_all = TRUE) %>%
  dplyr::select(c(1:18, 25))

#Retain information regarding assignment of genes to specific GO Terms with evidence codes
ASDGenes.in.SigGOIDs <- dplyr::select(ASDGenes.in.SigGOIDs, EntrezGeneID, Gene.Name, UniProtID, GO.BiologicalProcess=GOALL, GO.Evidence=EVIDENCEALL, EnsID=ENSEMBL)

#Remove data from environment that are no longer neccesary
rm(ASD_GOBPdata)
rm(ASDgene_BPGSA_classicfisher)
rm(ASDgene_BPGSA_classicfisher.results)
rm(ASDensIDs)
rm(ASDbraingenes)
rm(Genes.in.SigGOIDs)
rm(all_genes)
rm(geneUniverse)
rm(genesOfInterest)
rm(ASDgeneBPs)
###Add comlumn indicating to which gene ontology categories each gene is assigned?
```

4) Evaluate if proteins encoded by ASD risk genes are in a protein-protein interaction network with proteins encoded by ACMG genes
``` {r eval=TRUE}
#Load current list of recommended genes and convert to Ensembl IDs
ACMGgenelist <- read.table(url("https://github.com/veatcho/EffGeneXlaCDS/blob/master/ACMGlist.txt?raw=true"), header = FALSE, sep = "\t", stringsAsFactors = FALSE)
ACMGgenelist <- mutate(ACMGgenelist, external_gene_name=V1)
ACMGgenelist <- inner_join(ACMGgenelist, ensIDswithproteins_distinctgenes, by="external_gene_name")

#Export list for PPI analysis
ACMGgenelistforPPI <- dplyr::select(ACMGgenelist, ensembl_gene_id)
ASDensIDsforPPI <- dplyr::select(ASDensIDs_gene.names, ensembl_gene_id)
ASD_ACMGlistforPPI <- rbind(ASDensIDsforPPI, ACMGgenelistforPPI)
ASD_ACMGlistforPPI <- distinct(ASD_ACMGlistforPPI, ensembl_gene_id)
write.csv(ASD_ACMGlistforPPI, file="ASD_ACMGlistforPPI.csv")

#Ran PPI network builder on exported list of genes in https://www.networkanalyst.ca/faces/home.xhtml using a Generic PPI and IMEx Interactome data
ASD_ACMGPPINetwork <- read.table(url("https://github.com/veatcho/EffGeneXlaCDS/blob/master/ACMG_ASDnetwork1.sif?raw=true"), header = FALSE, sep = "\t", stringsAsFactors = FALSE)

#Determine how many proteins encoded by ASD risk genes are predicted to directly interact with proteins encoded by ACMG genes
ASDgenelist <- ASDensIDs_gene.names %>%
  mutate(V1=external_gene_name) %>% 
  mutate(V3=V1) %>%
  dplyr::select(V1, V3)

#Create list of protein 1 connections
ASD_ACMGPPINetwork_1ACMG <- semi_join(ASD_ACMGPPINetwork, ACMGgenelist, by='V1')
ASD_ACMGPPINetwork_1ACMG <- dplyr::select(ASD_ACMGPPINetwork_1ACMG, c(V1, V3))
ASD_ACMGPPINetwork_1ACMGtoASD <-  semi_join(ASD_ACMGPPINetwork_1ACMG, ASDgenelist, by='V3')
ASD_ACMGPPINetwork_1ASD <- semi_join(ASD_ACMGPPINetwork, ASDgenelist, by='V1')
ASD_ACMGPPINetwork_1ASD <- dplyr::select(ASD_ACMGPPINetwork_1ASD, c(V1, V3))
ACMGgenelist <- ASD_ACMGPPINetwork_1ASD %>%
  mutate(V3=V1)
ASD_ACMGPPINetwork_1ASDtoACMG <- semi_join(ASD_ACMGPPINetwork_1ASD, ACMGgenelist, by='V3')
ASD_ACMGPPINetwork_1 <- rbind.data.frame(ASD_ACMGPPINetwork_1ACMGtoASD, ASD_ACMGPPINetwork_1ASDtoACMG)

#Create list of protein 2 connections
ASD_ACMGPPINetwork_2ACMG <- semi_join(ASD_ACMGPPINetwork, ACMGgenelist, by='V3')
ASD_ACMGPPINetwork_2ACMG <- dplyr::select(ASD_ACMGPPINetwork_2ACMG, c(V1, V3))
ASD_ACMGPPINetwork_2ASDtoACMG <-  semi_join(ASD_ACMGPPINetwork_2ACMG, ASDgenelist, by='V1')
ASD_ACMGPPINetwork_2ASD <- semi_join(ASD_ACMGPPINetwork, ASDgenelist, by='V3')
ASD_ACMGPPINetwork_2ASD <- dplyr::select(ASD_ACMGPPINetwork_2ASD, c(V1, V3))
ASD_ACMGPPINetwork_2ACMGtoASD <- semi_join(ASD_ACMGPPINetwork_2ASD, ACMGgenelist, by='V1')
ASD_ACMGPPINetwork_proteinsofinterest <- rbind.data.frame(ASD_ACMGPPINetwork_1, ASD_ACMGPPINetwork_2ACMGtoASD, ASD_ACMGPPINetwork_2ASDtoACMG)

#Identify ASD genes that encode proteins that are recommended to test by the ACMG or directly interact with ACMG gene proteins
ASDGenes.Brain.SigBPs$ACMGGene<- ASDGenes.Brain.SigBPs$Gene.Name %in% ACMGgenelist$V1

ASDGenes.Brain.SigBPs$ACMG.PPI<- ASDGenes.Brain.SigBPs$Gene.Name %in% ASD_ACMGPPINetwork_proteinsofinterest$V1 | ASDGenes.Brain.SigBPs$Gene.Name %in% ASD_ACMGPPINetwork_proteinsofinterest$V3

##Determine how many ACMG proteins with which each ASD protein interacts?

#Remove data from environment that are no longer neccesary
rm(ensIDswithproteins_distinctgenes)
rm(ACMGgenelist)
rm(ASDgenelist)
rm(ACMGgenelistforPPI)
rm(ASDensIDsforPPI)
rm(ASD_ACMGlistforPPI)
rm(ASD_ACMGPPINetwork)
rm(ASD_ACMGPPINetwork_1ACMG)
rm(ASD_ACMGPPINetwork_1ACMGtoASD)
rm(ASD_ACMGPPINetwork_1ASD)
rm(ASD_ACMGPPINetwork_1ASDtoACMG)
rm(ASD_ACMGPPINetwork_1)
rm(ASD_ACMGPPINetwork_2ACMG)
rm(ASD_ACMGPPINetwork_2ASDtoACMG)
rm(ASD_ACMGPPINetwork_2ASD)
rm(ASD_ACMGPPINetwork_2ACMGtoASD)
```

5) These genes were then be prioritized based on an mouse phenotype term that reflects symptoms in ASD when manipulated in mice, by querying the genotype-phenotype representational state transfer (REST) application program interface (API) from the International Mouse Phenotyping Consortium (IMPC; http://www.mousephenotype.org/). The goal was to identify ASD risk genes that when knocked out of mice have a phenotypic consequence (associated with p<0.05) that is comparable to top Gene Ontology-defined biological processes overrepresented for ASD risk genes. Details for requesting mouse-phenotype data can be found here: http://www.mousephenotype.org/data/documentation/data-access 
``` {r eval=TRUE}
library('jsonlite')
#Identify genes that when knocked out of mice are associated (p<0.05) with ASD-related symptoms
#Note: must pull 'top level terms' from IMPC

AbnormalNervousSystemMouse0.05 <- fromJSON('https://www.ebi.ac.uk/mi/impc/solr/genotype-phenotype/select?q=top_level_mp_term_id:%22MP:0003631%22AND%20p_value:%5b0%20TO%200.05%5d&rows=100000&wt=json&indent=1', flatten = TRUE)

AbnormalBehaviorNeuroMouse0.05 <- fromJSON('https://www.ebi.ac.uk/mi/impc/solr/genotype-phenotype/select?q=top_level_mp_term_id:%22MP:0005386%22AND%20p_value:%5b0%20TO%200.05%5d&rows=100000&wt=json&indent=1')

#Reformat lists to pull mouse gene names and phenotypic consequences in KOs 
AbnormalNervousSystemMouseGenes<-data.frame(phenotype=AbnormalNervousSystemMouse0.05$response$docs$mp_term_name[c(1:532)],  genes=AbnormalNervousSystemMouse0.05$response$docs$marker_symbol[c(1:532)], stringsAsFactors = FALSE)

AbnormalBehaviorNeuroMouseGenes<-data.frame(phenotype=AbnormalBehaviorNeuroMouse0.05$response$docs$mp_term_name[c(1:7025)],  genes=AbnormalBehaviorNeuroMouse0.05$response$docs$marker_symbol[c(1:7025)], stringsAsFactors = FALSE)

MouseKOASDtraits<-distinct(full_join(AbnormalNervousSystemMouseGenes, AbnormalBehaviorNeuroMouseGenes, by='genes'))

#Identify the most likely human orthologs of mouse genes identified above
#The mouse gene list was exported and batch searched using all prediction tools available via the DRSC Integrative Ortholog Prediction Tool (DIOPT; Version 7.1 March 2018; https://www.flyrnai.org/diopt) with filtering to return only best match when there is more than one match per input gene or protein
MouseKOASDtraitsgenes <- distinct(MouseKOASDtraits, genes)
write(MouseKOASDtraitsgenes$genes, file="MouseKOASDtraitsgenes.txt")

#Pull results from mouse/human ortholog query using DIOPT
MouseKOASDtraitsgenes.hsortho <- read.table(url("https://github.com/veatcho/EffGeneXlaCDS/blob/master/MouseKO_ASDtraitsgenes_hsortho.xls?raw=true"), header = TRUE, sep = "\t", fill = TRUE, stringsAsFactors = FALSE)

#Identify ASD risk genes expressed in human brain that also have a relevant ASD-related phenotype when knocked out of mice
ASDGenes.Brain.SigBPs <- mutate(ASDGenes.Brain.SigBPs, Human.GeneID = EntrezGeneID)
ASDGenes.Brain.SigBPs.MouseKOs <- left_join(ASDGenes.Brain.SigBPs, MouseKOASDtraitsgenes.hsortho[, c(1, 6)], by='Human.GeneID')
MouseKOASDtraits <- mutate(MouseKOASDtraits, Mouse.GeneID=MouseKOASDtraits$genes)
ASDGenes.Brain.SigBPs.MouseKOPhenotypes <- left_join(ASDGenes.Brain.SigBPs.MouseKOs, MouseKOASDtraits, by='Mouse.GeneID')

#Determine how many ASD-related phenotypes were associated with knocking out each ASD risk gene
ASDGenes.Brain.SigBPs.MouseKOs <- ASDGenes.Brain.SigBPs.MouseKOPhenotypes %>%
  group_by(EntrezGeneID) %>%
  mutate(nMouseKOPhenotypeX = n_distinct(phenotype.x), nMouseKOPhenotypeY = n_distinct(phenotype.y), nMouseKOTraits=(nMouseKOPhenotypeX+nMouseKOPhenotypeY)) %>%
  distinct(UniProtID, .keep_all = TRUE) %>%
  dplyr::select(c(1:23, 29))

#Retain information regarding assignment of genes to specific GO Terms with evidence codes
ASDGenes.Brain.SigBPs.MouseKOPhenotypes <- ASDGenes.Brain.SigBPs.MouseKOPhenotypes %>%
  filter(!is.na(Mouse.GeneID)) %>%
  dplyr::select(c(1:3), AbnormalNervousSystem.MouseKOtrait=phenotype.x, AbnormalBehaviorNeuro.MouseKOtrait=phenotype.y, 26)

#Remove data from environment that are no longer neccesary
rm(AbnormalBehaviorNeuroMouse0.05)
rm(AbnormalBehaviorNeuroMouseGenes)
rm(AbnormalNervousSystemMouse0.05)
rm(AbnormalNervousSystemMouseGenes)
rm(MouseKOASDtraits)
rm(MouseKOASDtraitsgenes)
rm(MouseKOASDtraitsgenes.hsortho)
```

6) To identify currently approved drugs that target or have drug ontology to proteins encoded by the prioritized genes implicated in ASD, a software request was sent via RCurl to the Pharos API (https://pharos.nih.gov/idg/dto).
``` {r eval=TRUE}
#Details for pulling from Pharos can be found here: https://pharos.nih.gov/idg/api
#Note this is one of the pulls that takes the longest! Would love suggestions for reducing time to pull these data
targets_pharos <- fromJSON("https://pharos.nih.gov/idg/api/v1/targets?facet=IDG+Development+Level&top=25000&wt=json&indent=1")
drugtargets<-data.frame(genes=targets_pharos$content$gene[c(1:20244)], idgTDL=targets_pharos$content$idgTDL[c(1:20244)], uniprotId=targets_pharos$content$accession[c(1:20244)], stringsAsFactors = FALSE)

#Identify ASD risk genes expressed in human brain, have a relevant ASD-related phenotype when knocked out of mice and are, or have ontology to, FDA-approved drug targets
drugtargets <- drugtargets %>%
  dplyr::select(Gene.Name=genes, idgTDL, UniProtID=uniprotId)

ASDGenes.Brain.SigBPs.MouseKOs.DrugTargets <- left_join(ASDGenes.Brain.SigBPs.MouseKOs, drugtargets, by='Gene.Name')
ASDGenes.Brain.SigBPs.MouseKOs.DrugTargets <- distinct(ASDGenes.Brain.SigBPs.MouseKOs.DrugTargets, EntrezGeneID, .keep_all = TRUE)
  
#Output final dataframe with attributes included
ASDRiskGeneAttributes <- write.csv(ASDGenes.Brain.SigBPs.MouseKOs.DrugTargets, file="ASDRiskGeneAttributes.csv")

#output additional files to aid in more detailed desrcriptions of gene attributes

#Remove data from environment that are no longer neccesary
rm(drugtargets)
rm(targets_pharos)
```

7) PharmGKB data were reviewed (no API available yet) to identify evidence of genes with variants that influence individual response to drugs

``` {r eval=TRUE}
#Work in progress
```


9) VALIDATION APPROACHES (additional thoughts are appreciated): Evaluate if the annotations observed for the 'ASD gene set' were unique to ASD risk genes by evaluating the same annotations in a random sample of human protein coding genes, equal to the number of ASD risk genes pulled from DisGeNET (n=917).
```{r eval=TRUE}
```    
